matrix/Quaternion.hpp

```diff
@@ -280,9 +280,11 @@ class Quaternion : public Vector<Type, 4>
     void invert()
     {
         Quaternion &q = *this;
-        q(1) *= -1;
-        q(2) *= -1;
-        q(3) *= -1;
+        Type n = q.dot(q);
```

Comment:
Here is the derivation with our quaternion product. You cannot assume unit length here. This derivation works for unit lenght and non-unit lenght quaternions: https://github.com/jgoppert/iekf_analysis/blob/master/Quaternion%20Inverse%20Check.ipynb